src/ATen/function_wrapper.py

```diff
@@ -761,6 +775,8 @@ def emit_body(env, option):
         body = []
         body += handle_sparse(env, option)
         body += handle_zero_dim(env, option)
+        if only_zero_dim(env, option):
```

Comment:
Why does this cause a return?

Reply:
the generated code is:
```
Tensor & CPUFloatType::m_fill_(Tensor & self, const Tensor & value) const {
    if(value.dim() == 0) {
        return static_cast<const Type*>(this)->m_fill_(self, Scalar(value));
    }
    else {
        runtime_error("fill_ only supports a 0-dimensional value tensor, but got tensor "
                      "with %" PRId64 " dimension(s)", value.dim());
    }
}
```

so anything else is unreachable.  Also, the further generated code if we didn't return would be invalid as of now; we could change it to not be, but since it's unreachable it didn't seem worth the effort.

Reply:
ok I just have some small nits about this to make it more understandable:
I think we should avoid templates that are not complete statements/expressions, so instead of `else ...`, can we just repeat the check `if(what.dim() != 0) { ... }`. This will be more robust to future modifications. 

Can we also make all the only_zero_dim handling happen in one function? `only_zero_dim` -> `handle_only_zero_dim`, and merge all the only emission code  into it:

```
zero_dim_check = handle_only_zero_dim(env, option)
if zero_dim_check is not None:
   # code below zero dim check is an asset so we do not need to generate the rest.
   body += zero_dim_check
   return
```